### PR TITLE
DAOS-1968 control: auto generate config file

### PR DESCRIPTION
**Pull request by tanabarr**
Updates to initial implementation:
- remove duplicate NetDevClass consts and instead import from netdetect
- refactor auto config logic for clarity and improve comments
- display Info on host errors in tabular form
- take access points on the dmg config generate commandline

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/lib/netdetect/netdetect.go [hunk 1]
```diff
@@ -1303,6 +1306,8 @@ func DevClassName(class uint32) string {
 		return "EUI64"
 	case Infiniband:
 		return "INFINIBAND"
+	case NetDevAny:
+		return "best-available"
```
Comment (kjacque) [inline src/control/lib/netdetect/netdetect.go:1310]: NetDevAny seems like it really belongs further up the stack, maybe in lib/control, seeing that it's not actually used in the netdetect lib outside of this DevClassName function. As it is, this string seems very tightly coupled to DMG.

Comment (tanabarr) [reply src/control/lib/netdetect/netdetect.go:1310]: initially , I did have it in the control API but it felt strange there, I can move it back but I'm not sure it's any better

**Comment by tanabarr** [comment]
gatekeeper please use PR description and title when landing

**Comment by tanabarr** [comment]
example workflow:
```
[tanabarr@wolf-118 daos-control-demo]$ sudo su daos_server                                 bash-4.2$ dmg storage scan
Hosts          SCM Total             NVMe Total
-----          ---------             ----------
wolf-[118-121] 6.4 TB (2 namespaces) 3.5 TB (4 controllers)
bash-4.2$ dmg network scan
--------------
wolf-[118-121]
--------------

    -------------
    NUMA Socket 0
    -------------

        Provider          Interfaces
        --------          ----------
        ofi+tcp;ofi_rxm   ib0, eth0
        ofi+verbs         ib0
        ofi+tcp           ib0, eth0
        ofi+sockets       ib0, eth0
        ofi+psm2          ib0
        ofi+verbs;ofi_rxm ib0, eth0

    -------------
    NUMA Socket 1
    -------------

        Provider          Interfaces
        --------          ----------
        ofi+psm2          ib1
        ofi+verbs;ofi_rxm ib1
        ofi+tcp;ofi_rxm   ib1
        ofi+verbs         ib1
        ofi+tcp           ib1
        ofi+sockets       ib1

bash-4.2$ dmg config generate

port: 10001
transport_config:
  allow_insecure: false
  server_name: server
  client_cert_dir: /etc/daos/certs/clients
  ca_cert: /etc/daos/certs/daosCA.crt
  cert: /etc/daos/certs/server.crt
  key: /etc/daos/certs/server.key
servers:
- targets: 16
  nr_xs_helpers: 2
  first_core: 0
  name: daos_server
  socket_dir: /var/run/daos_server
  log_file: /tmp/daos_io_server.0.log
  scm_mount: /mnt/daos0
  scm_class: dcpm
  scm_list:
  - /dev/pmem0
  bdev_class: nvme
  bdev_list:
  - 0000:5e:00.0
  - 0000:5f:00.0
  provider: ofi+psm2
  fabric_iface: ib0
  fabric_iface_port: 31416
  pinned_numa_node: 0
- targets: 16
  nr_xs_helpers: 2
  first_core: 0
  name: daos_server
  socket_dir: /var/run/daos_server
  log_file: /tmp/daos_io_server.1.log
  scm_mount: /mnt/daos1
  scm_class: dcpm
  scm_list:
  - /dev/pmem1
  bdev_class: nvme
  bdev_list:
  - 0000:81:00.0
  - 0000:da:00.0
  provider: ofi+psm2
  fabric_iface: ib1
  fabric_iface_port: 32416
  pinned_numa_node: 1
disable_vfio: false
disable_vmd: true
nr_hugepages: 0
set_hugepages: false
control_log_mask: INFO
control_log_file: /tmp/daos_server.log
helper_log_file: ""
firmware_helper_log_file: ""
recreate_superblocks: false
fault_path: ""
name: daos_server
socket_dir: /var/run/daos_server
provider: ofi+psm2
modules: ""
access_points:
- localhost:10001
fault_cb: ""
hyperthreads: false
path: ../etc/daos_server.yml
```

**Comment by tanabarr** [comment]
error reporting:
```
bash-4.2$ dmg config generate -l wolf-[118-121,130-133]
Errors:
  Hosts    Error
  -----    -----
  wolf-130 the server at wolf-130:10001 refused the connection
  wolf-131 the server at wolf-131:10001 refused the connection
  wolf-132 the server at wolf-132:10001 refused the connection
  wolf-133 the server at wolf-133:10001 refused the connection

ERROR: dmg: 4 hosts had errors
bash-4.2$ dmg config generate -l wolf-[118-121,71]
Heterogeneous storage hardware configurations detected, cannot proceed. The following sets of hosts have different storage hardware:
wolf-71:10001
wolf-[118-121]:10001

ERROR: dmg: storage hardware not consistent across hosts
```

File: src/control/cmd/dmg/auto.go [hunk 2]
```diff
@@ -64,31 +68,41 @@ func (cmd *configGenCmd) Execute(_ []string) error {
...
 
-	if err != nil {
+	resp, err := control.ConfigGenerate(ctx, req)
+
+	// host level errors e.g. unresponsive daos_server process
+	var bld strings.Builder
+	if err := pretty.PrintResponseErrors(resp, &bld); err != nil {
 		return err
 	}
+	cmd.log.Info(bld.String()) // no-op if no host level errors
```
Comment (mjmac) [inline src/control/cmd/dmg/auto.go:93]: Shouldn't this be logged at Error level?

**Review by mjmac — COMMENTED** [review]
Looks fine, just one question.

**Review by kjacque — APPROVED** [review]

File: src/control/cmd/dmg/auto.go (hunk 2 above)
Comment (tanabarr) [reply src/control/cmd/dmg/auto.go:93]: yes it should, mind if I move this fix to my following PR? I don't think it's a blocking change

**Comment by mjmac** [comment]
Known failures:
  - [DAOS-5749](https://jira.hpdd.intel.com/browse/DAOS-5749)
  - [DAOS-5992](https://jira.hpdd.intel.com/browse/DAOS-5992)

**Pull request merged**